lib/mocha.js

```diff
@@ -994,6 +1033,65 @@ Mocha.prototype.rootHooks = function rootHooks(hooks) {
   return this;
 };
 
+/**
+ * Toggles parallel mode.
+ *
+ * Must be run before calling {@link Mocha#run}. Changes the `Runner` class to
+ * use; also enables lazy file loading if not already done so.
+ * @param {boolean} [enable] - If `true`, enable; otherwise disable.
+ * @throws If run in browser
+ * @throws If Mocha not in "INIT" state
+ * @returns {Mocha}
+ * @chainable
+ * @public
+ */
+Mocha.prototype.parallelMode = function parallelMode(enable) {
+  if (utils.isBrowser()) {
+    throw errors.createUnsupportedError(
+      'parallel mode is only supported in Node.js'
+    );
+  }
+  var parallel = enable === true;
+  if (
+    parallel === this.options.parallel &&
+    !this._lazyLoadFiles &&
+    this._runnerClass !== exports.Runner
+  ) {
+    return this;
+  }
+  if (this._state !== mochaStates.INIT) {
```

Comment:
Note: the state of the mocha instance is _reset_ to `INIT` after a run (if `cleanReferencesAfterRun` is `false`). So a user can change the `parallelMode` after mocha has run. 

This is fine, as long as we implement the `dispose` on the `bufferedRunner` correctly. It would mean that the parallel runner / buffered runner is disposed and a new one is started. We might want to change this behavior in the future, but it's fine for now I would say.